editor/filesystem_dock.cpp

```diff
@@ -1601,7 +1601,9 @@ void FileSystemDock::_rename_operation_confirm() {
 		EditorNode::get_singleton()->show_warning(TTR("This filename begins with a dot rendering the file invisible to the editor.\nIf you want to rename it anyway, use your operating system's file manager."));
 		rename_error = true;
 	} else if (to_rename.is_file && to_rename.path.get_extension() != new_name.get_extension()) {
-		if (!EditorFileSystem::get_singleton()->get_valid_extensions().find(new_name.get_extension())) {
+		if(new_name.get_extension()==""){// no extention provided
```

Comment:
You need to follow the [code style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html)

Reply:
It's best to use `clang-format`.

Reply:
More than just the code style, the empty check should be done with `is_empty()`.